### PR TITLE
apply simple cache for method in arc/orc

### PR DESCRIPTION
**Pull request by ringabout**
ref https://github.com/nim-lang/Nim/issues/18612

### CODE REVIEW

File: lib/system/arc.nim [hunk 1]
```diff
@@ -227,11 +227,37 @@ template tearDownForeignThreadGc* =
...
+  else:
+    cache[0] = objName
+    result = false
+
+proc isSameCstring(x, y: cstring): bool {.inline.} =
+  proc strcmp(a, b: cstring): cint {.noSideEffect,
+    importc, header: "<string.h>".}
+  if pointer(x) == pointer(y): result = true
+  elif x.isNil or y.isNil: result = false
+  else: result = strcmp(x, y) == 0
```
Comment (Araq) [inline lib/system/arc.nim:250]: Does this `strcmp` pay off? Because it feels like it nullifies your entire optimization.

Comment (ringabout) [reply lib/system/arc.nim:250]: It seems to work based on the benchmark of https://github.com/nim-lang/Nim/issues/18612#issuecomment-890531425

I will examine further.

Comment (ringabout) [reply lib/system/arc.nim:250]: It seems that strcmp is not needed => https://github.com/nim-lang/Nim/pull/19749

**Comment by ringabout** [comment]
Succeeded by https://github.com/nim-lang/Nim/pull/19749

**Pull request closed without merging**